scipy/linalg/tests/test_lapack.py

```diff
@@ -3299,40 +3299,92 @@ def test_gges_tgsen(dtype):
+                 [ -7.0,  7.0, -6.0, 19.0]]),
+      np.array( [[  2.0,  1.0,  1.0,  3.0] ,
+                 [  0.0,  1.0,  2.0,  1.0] ,
+                 [  0.0,  0.0,  1.0,  1.0] ,
+                 [  0.0,  0.0,  0.0,  2.0]]),
+      np.array( [[  1.0,  1.0,  1.0,  2.0] ,
+                 [  0.0,  1.0,  4.0,  1.0] ,
+                 [  0.0,  0.0,  1.0,  1.0] ,
+                 [  0.0,  0.0,  0.0,  1.0]]),
+      np.array( [[ -7.0,  5.0,  0.0,  7.0] ,
+                 [ -5.0,  1.0, -8.0,  0.0] ,
+                 [ -1.0,  2.0, -3.0,  5.0] ,
+                 [ -3.0,  2.0,  0.0,  5.0]]),
+      np.array( [[  1.0,  1.0,  1.0,  1.0] ,
+                 [ -1.0,  2.0, -1.0, -1.0] ,
+                 [ -1.0,  1.0,  3.0,  1.0] ,
+                 [ -1.0,  1.0, -1.0,  4.0]]),
+     np.array(  [[  4.0, -1.0,  1.0, -1.0] ,
+                 [  1.0,  3.0, -1.0,  1.0] ,
+                 [ -1.0,  1.0,  2.0, -1.0] ,
+                 [  1.0, -1.0,  1.0,  1.0]]) )])
 @pytest.mark.parametrize('dtype', REAL_DTYPES)
-@pytest.mark.parametrize('trans', ('N'))#, 'T'
-def test_tgsyl(dtype,trans):
-    
-    seed(1234)
-    atol = np.finfo(dtype).eps*1e4
+def test_tgsyl_NAG(a,b,c,d,e,f,r,l,dtype):
+    atol = 1e-4
     
-    m, n = 5, 3
+    tgsyl = get_lapack_funcs(('tgsyl'), dtype=dtype)
+    rans,lans,scale,dif,info = tgsyl(a,b,c,d,e,f)
     
-    a = np.triu(generate_random_dtype_array([m, m], dtype=dtype))
-    d = np.triu(generate_random_dtype_array([m, m], dtype=dtype))
-    b = np.triu(generate_random_dtype_array([n, n], dtype=dtype))
-    e = np.triu(generate_random_dtype_array([n, n], dtype=dtype))
-    c = generate_random_dtype_array([m, n], dtype=dtype)
-    f = generate_random_dtype_array([m, n], dtype=dtype)
-
-    tgsyl, tgsyl_lwork = get_lapack_funcs(
-        ('tgsyl','tgsyl_lwork'), dtype=dtype)
-    ijob =1
-    lwork = _compute_lwork(tgsyl_lwork,m, n)#, trans=trans, ijob=ijob)
-
-    print(lwork)
-    # off-by-one error in LAPACK, see gh-issue #13397
-    # lwork = (lwork[0]+1, lwork[1])
-
-    r,l,scale,dif,iwork,info = tgsyl(a,b,c,d,e,f,trans=trans,ijob=ijob)
     assert_equal(info, 0)
-    lhs = list(np.matmul(a,r)-np.matmul(l,b))
-    rhs = scale*c
-    assert_allclose(lhs,rhs,rtol=1e-6,atol=1e-6)
-
-    lhs = list(np.matmul(d,r)-np.matmul(l,e))
-    rhs = scale*f
-    assert_allclose(lhs,rhs,rtol=1e-6,atol=1e-6)
-
-    # assert_allclose(dif,0.0,rtol=1e-6,atol=1e-6)
-
+    assert_almost_equal(scale,1.0,decimal=4,err_msg="SCALE must be 1.0")
+    assert_almost_equal(dif,0.0,decimal=2,err_msg="DIF must be nearly 0")
+    assert_allclose(rans,r,atol=atol,err_msg='Solution for unknown R of Sylvester equation is not correct')
+    assert_allclose(lans,l,atol=atol,err_msg='Solution for unknown L of Sylvester equation is not correct')
+    
+@pytest.mark.parametrize('dtype', REAL_DTYPES)
+@pytest.mark.parametrize('trans', ('N', 'T'))
+@pytest.mark.parametrize('ijob', [0,1,2,3,4])
+def test_tgsyl(dtype,trans,ijob):
```

Comment:
@ilayn @Kai-Striega I can't see why the precision is so poor. I can of course, increase atol and rtol further or compare results upto certain decimals, but I'm a bit circumspect in doing so. Let me know if you see anything obvious in the test code. The only thing which maybe causing a problem is that the element(0,0) of a and b are larger compared to others since I used schur to get them into the quasi triangular form. An example A matrix looks like this: 
![image](https://github.com/scipy/scipy/assets/34144786/6f7a1cac-2f6f-4b58-ab97-8e832cf80a39)